app/api/entities/v2_support.ts

```diff
@@ -65,23 +64,16 @@ const deleteRelatedNewRelationships = async (sharedId: string) => {
   }
 };
 
-const markNewRelationshipsOfAffected = async (
+const denormalizeAfterEntityCreation = async (
   { sharedId, language }: { sharedId: string; language: string },
   index: boolean = true
 ) => {
   const transactionManager = new MongoTransactionManager(getClient());
   if (await DefaultSettingsDataSource(transactionManager).readNewRelationshipsAllowed()) {
-    const entitiesDataSource = DefaultEntitiesDataSource(transactionManager);
-    const service = DenormalizationService(transactionManager);
-    const candidates = await service.getCandidateEntitiesForEntity(sharedId, language);
-
-    await entitiesDataSource.markMetadataAsChanged(candidates);
-
-    if (index && candidates.length > 0) {
-      await search.indexEntities(
-        { sharedId: { $in: candidates.map(c => c.sharedId) } },
-        '+fullText'
-      );
+    const denormalizationService = DenormalizationService(transactionManager);
```

Comment:
Let's discuss this part. What I don't like is that the default DenormalizationService receives a transactionManager (and a mongo one, even), but I see that the reason maybe that you need to run that executeOnCommitHandlers. So let's see if we can figure something out.